src/Service/ParcelShopsService.php

```diff
@@ -24,20 +25,22 @@ public function __construct(Client $client, Serializer $serializer)
     }
 
     /**
-     * @throws BadRequestException
+     * @return ParcelShop[]
+     *
      * @throws ServiceUnavailableException
+     * @throws BadRequestException
      */
-    public function get(): ParcelShopsResult
+    public function get(): array
     {
         $response = $this->client->request(new GetParcelShops());
 
-        /** @var ParcelShopsResult $parcelShopsResult */
-        $parcelShopsResult = $this->serializer->decodeResponse(ParcelShopsResult::class, $response);
+        /** @var ParcelShopsResponse $parcelShopsResult */
+        $parcelShopsResult = $this->serializer->decodeResponse(ParcelShopsResponse::class, $response);
 
         if (!$parcelShopsResult->getStatus()->is(StatusResult::ok())) {
-            throw new BadRequestException('Invalid request.', $parcelShopsResult);
+            throw new BadRequestException($parcelShopsResult);
         }
 
-        return $parcelShopsResult;
+        return $parcelShopsResult->getData();
```

Comment:
if all is ok I simply return array of parcel shops instead of all response object.
If error occurs throw exception.